pkg/webhook/admission/defaulter_custom.go

```diff
@@ -21,29 +21,48 @@ import (
 	"encoding/json"
 	"errors"
 	"net/http"
+	"slices"
 
+	"gomodules.xyz/jsonpatch/v2"
 	admissionv1 "k8s.io/api/admission/v1"
 	apierrors "k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/sets"
 )
 
 // CustomDefaulter defines functions for setting defaults on resources.
 type CustomDefaulter interface {
 	Default(ctx context.Context, obj runtime.Object) error
 }
 
+type defaulterOptions struct {
+	removeUnknownFields bool
+}
+
+type defaulterOption func(*defaulterOptions)
+
+// DefaulterRemoveUnknownFields makes the defaulter prune the fields that are not recognized in the local scheme.
+func DefaulterRemoveUnknownFields(o *defaulterOptions) {
+	o.removeUnknownFields = true
+}
+
 // WithCustomDefaulter creates a new Webhook for a CustomDefaulter interface.
-func WithCustomDefaulter(scheme *runtime.Scheme, obj runtime.Object, defaulter CustomDefaulter) *Webhook {
+func WithCustomDefaulter(scheme *runtime.Scheme, obj runtime.Object, defaulter CustomDefaulter, opts ...defaulterOption) *Webhook {
```

Comment:
Can we also please add a variadic arg for this into ` *WebhookBuilder) WithDefaulter`? Right now its not possible to configure this from there

Reply:
done